Code/GraphMol/Substruct/catch_tests.cpp

```diff
@@ -587,14 +589,18 @@ TEST_CASE("pickling HasPropWithValue queries") {
       REQUIRE(pklmol.getAtomWithIdx(0)->hasQuery());
       REQUIRE(pklmol.getAtomWithIdx(0)->hasQuery());
       CHECK(SubstructMatch(*target, pklmol, ps).size() == 0);
+      // make sure we are idempotent in pickling
+      CHECK(SubstructMatch(*target, mol2, ps).size() == 0);
     }
     {
       std::string pkl;
-      MolPickler::pickleMol(mol2, pkl);
+      MolPickler::pickleMol(mol3, pkl);
       RWMol pklmol(pkl);
       REQUIRE(pklmol.getAtomWithIdx(0)->hasQuery());
       REQUIRE(pklmol.getAtomWithIdx(0)->hasQuery());
       CHECK(SubstructMatch(*target, pklmol, ps).size() == 0);
+      // make sure we are idempotent in pickling
+      CHECK(SubstructMatch(*target, mol3, ps).size() == 0);
```

Comment:
What are you testing with these additions?

Reply:
This is ensuring that pickling doesn't change the original molecule.  I forget exactly why I added it, but I think I had seen a case where pickling changed the results when the original molecule was used again.

Reply:
The only way pickling would change the substructure matching behavior in this case is if the properties change, and that's a test for the pickling suite, not here.

Reply:
This is where we had added the pickle tests before, I'll move them.

Reply:
> This is where we had added the pickle tests before, I'll move them.

There's no need to move anything, it's not like extra tests are bad, but I don't think these are actually testing anything real

Reply:
With the change to PairHolder this shouldn't happen anymore.